include/envoy/router/rds.h

```diff
@@ -23,7 +23,32 @@ class RouteConfigProvider {
   virtual Router::ConfigConstSharedPtr config() PURE;
 };
 
+/**
+ * A provider for dynamic route configurations.
+ */
+class RdsRouteConfigProvider : public RouteConfigProvider {
+public:
+  virtual ~RdsRouteConfigProvider() {}
+
+  /**
+   * @return std::string the loaded route table.
+   */
+  virtual std::string configAsString() PURE;
+
+  /**
+   * @return std::string the name of the configured route table.
+   */
+  virtual std::string routeConfigName() PURE;
```

Comment:
Should some of these be `std::string&`? This is just a nit, it might make sense for the implementation to have the freedom to not maintain a const string here.